test/e2e/storage/csi_volumes.go

```diff
@@ -219,18 +155,16 @@ var _ = utils.SIGDescribe("CSI Volumes [Flaky]", func() {
 
 		BeforeEach(func() {
 			By("deploying csi hostpath driver")
-			clusterRole = csiClusterRole(cs, config, false)
-			serviceAccount = csiServiceAccount(cs, config, false)
-			csiClusterRoleBinding(cs, config, false, serviceAccount, clusterRole)
+			serviceAccount = csiCombinedServiceAccount(cs, config, false)
```

Comment:
this is up to @kubernetes/sig-storage-pr-reviews, but I'd recommend exercising each pair of phases (provision/delete, attach/detach, mount/unmount) using a separate service account with just the specified role. That way, e2e tests make it really clear when a new operation is performed in a phase that isn't covered by the relevant role.

Reply:
I think that would be a good test to have, however this test is for testing the HostPath CSI Plugin specifically, not the CSI external components. We should definitely add in tests for the external components soon @saad-ali.

@liggitt let me know if I misunderstood your comment

Reply:
> this test is for testing the HostPath CSI Plugin specifically, not the CSI external components

so does that mean it doesn't need the provisioner and attacher roles?

Reply:
The csi-provisioner, driver-registrar, and csi-attacher are side-car containers necessary for the HostPath CSI Plugin container. The pod containing all 4 containers will need provisioner, attacher, and driver-registrar roles.